cmake/sysbuild/suit.cmake

```diff
@@ -0,0 +1,283 @@
+#   suit_set_absolute_or_relative_path(<path> <relative_root> <output_variable>)
+#
+# Parameters:
+#   'path' - path to resolve
+#   'relative_root' - root folder used in case of relative path
+#   'output_variable' - variable to store results
+function(suit_set_absolute_or_relative_path path relative_root output_variable)
+  if(NOT IS_ABSOLUTE ${path})
+    file(REAL_PATH "${relative_root}/${path}" path)
+  endif()
+  set(${output_variable} "${path}" PARENT_SCOPE)
+endfunction()
+
+# Sign an envelope using SIGN_SCRIPT.
+#
+# Usage:
+#   suit_sign_envelope(<input_file> <output_file>)
+#
+# Parameters:
+#   'input_file' - path to input unsigned envelope
+#   'output_file' - path to output signed envelope
+function(suit_sign_envelope input_file output_file)
+  cmake_path(GET ZEPHYR_NRF_MODULE_DIR PARENT_PATH NRF_DIR_PARENT)
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT KCONFIG)
+  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT} ${NRF_DIR_PARENT} SIGN_SCRIPT)
+  if(NOT EXISTS ${SIGN_SCRIPT})
+    message(SEND_ERROR "DFU: ${CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT} does not exist. Corrupted configuration?")
+    return()
+  endif()
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SIGN_SCRIPT}
+    --input-file ${input_file}
+    --output-file ${output_file}
+  )
+endfunction()
+
+# Register SUIT post build commands.
+#
+# Usage:
+#   suit_register_post_build_commands()
+#
+function(suit_register_post_build_commands)
+  get_property(
+    post_build_commands
+    GLOBAL PROPERTY
+    SUIT_POST_BUILD_COMMANDS
+  )
+
+  foreach(image ${IMAGES})
+    sysbuild_get(BINARY_DIR IMAGE ${image} VAR APPLICATION_BINARY_DIR CACHE)
+    sysbuild_get(BINARY_FILE IMAGE ${image} VAR CONFIG_KERNEL_BIN_NAME KCONFIG)
+    list(APPEND dependencies "${BINARY_DIR}/zephyr/${BINARY_FILE}.bin")
+  endforeach()
+
+  add_custom_target(
+    create_suit_artifacts
+    ALL
+    ${post_build_commands}
+    DEPENDS
+    ${dependencies}
+    COMMAND_EXPAND_LISTS
+    COMMENT "Create SUIT artifacts"
+  )
+endfunction()
+
+# Create DFU package/main envelope.
+#
+# Usage:
+#   suit_create_package()
+#
+function(suit_create_package)
+  add_custom_target(
+    suit_prepare_output_folder
+    ALL
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${SUIT_ROOT_DIRECTORY}
+    COMMENT
+    "Create DFU output directory"
+  )
+  set(SUIT_OUTPUT_ARTIFACTS_ITEMS)
+  set(SUIT_OUTPUT_ARTIFACTS_TARGETS)
+  set(CORE_ARGS)
+  set(STORAGE_BOOT_ARGS)
+
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION KCONFIG)
+  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
+  sysbuild_get(ENVELOPE_SIGNATURE IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
+  if (NOT DEFINED ENVELOPE_SIGNATURE)
```

Comment:
`ENVELOPE_SIGNATURE` sounds like this would be the _content_ of the envelope signature, so I would rename this variable to something like `ENVELOPE_SHALL_BE_SIGNED`

Reply:
Good point, renamed.